.github/workflows/ci.yaml

```diff
@@ -0,0 +1,68 @@
+name: "CI"
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master 
+  create:
+    tags:
+      - v*
+
+defaults:
+  run:
+    shell: bash -l {0}
+
+jobs:
+  cancel:
+    name: "Cancel"
+    runs-on: ubuntu-latest
+    steps:
+      - name: "Get workflow id"
+        uses: styfle/cancel-workflow-action@0.6.0
+        with:
+          access_token: ${{ github.token }}
+
+  build:
+    name: "Build"
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os:
+          - ubuntu-latest
+          - macos-latest
+          - windows-latest
+    steps:
+
+      - name: Set up cache
+        uses: actions/cache@v2
+        with:
+          path: |
+            $HOME/.ghc
+            $HOME/.cabal
+            $HOME/.stack
+          key: ${{ runner.os }}-haskell
+
+      - name: Set up environment
+        uses: actions/setup-haskell@v1.1
```

Comment:
I couldn't find this action in Github marketplace but I found it on Github as a repo -> how is that?

Reply:
I have no idea... It looks like it's working fine in any case.